src/Transpiler.ts

```diff
@@ -335,12 +340,31 @@ export class LuaTranspiler {
         return result + this.indent + "end\n";
     }
 
+    public transpileLoopBody(
+        node: ts.WhileStatement
+            | ts.DoStatement
+            | ts.ForStatement
+            | ts.ForOfStatement
+            | ts.ForInStatement
+    ): string {
+        this.genVarCounter++;
+        this.loopStack.push(this.genVarCounter);
+        let result = this.indent + "do\n";
+        this.pushIndent();
+        result += this.transpileStatement(node.statement);
+        this.popIndent();
+        result += this.indent + "end\n";
+        result += this.indent + `::__continue${this.loopStack[this.loopStack.length - 1]}::\n`;
```

Comment:
You can just use this.loopStack.pop() instead of this.loopStack[this.loopStack.length - 1]. and remove the pop on the next line. pop will return the last value.